### PR TITLE
Smoke test framework in golang

### PR DESCRIPTION
**Pull request by ray-milkey**

### CODE REVIEW

File: cmd/smoke-test/smoketest.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+	for _, pathElement := range pathElements {
+		protoBuilder.WriteString("elem: <name: '")
+		protoBuilder.WriteString(pathElement)
+		protoBuilder.WriteString("'> ")
+	}
+	protoBuilder.WriteString(">")
+	return protoBuilder.String()
+}
+
+func makeSetPathProto(target string, path string, value string) string {
```
Comment (Andrea-Campanella) [inline cmd/smoke-test/smoketest.go]: you might want to consolidate this method and the one above. Avoid redundancy and error prone code

File: cmd/smoke-test/smoketest.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	protoBuilder.WriteString("', ")
+
+	pathElements := utils.SplitPath(path)
+	for _, pathElement := range pathElements {
+		protoBuilder.WriteString("elem: <name: '")
+		protoBuilder.WriteString(pathElement)
+		protoBuilder.WriteString("'> ")
+	}
+	protoBuilder.WriteString(">")
+	protoBuilder.WriteString(" val: <string_val: '")
```
Comment (Andrea-Campanella) [inline cmd/smoke-test/smoketest.go]: just add this to the builder.

File: cmd/smoke-test/smoketest.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+	protoBuilder.WriteString(value)
+	protoBuilder.WriteString("'>>")
+	return protoBuilder.String()
+}
+
+func findPathValue(response *gpb.GetResponse, path string) (string, error) {
+	if len(response.Notification) != 1 {
+		return "", errors.New("response notifications must have one entry")
```
Comment (Andrea-Campanella) [inline cmd/smoke-test/smoketest.go]: just out of my curiosity, why not fmt.Errorf ?

Comment (ray-milkey) [reply cmd/smoke-test/smoketest.go]: It should be. That seems to be the standard.

File: cmd/smoke-test/smoketest.go (hunk 1 above)
Comment (ray-milkey) [reply cmd/smoke-test/smoketest.go]: This was done in a subsequent set of changes

File: cmd/smoke-test/smoketest.go (hunk 3 above)
Comment (ray-milkey) [reply cmd/smoke-test/smoketest.go]: done

File: cmd/smoke-test/smoketest.go (hunk 2 above)
Comment (ray-milkey) [reply cmd/smoke-test/smoketest.go]: seems to have been refactored out

Comment (Andrea-Campanella) [reply cmd/smoke-test/smoketest.go]: +1

File: cmd/smoke-test/smoketest.go (hunk 1 above)
Comment (Andrea-Campanella) [reply cmd/smoke-test/smoketest.go]: nice!

File: cmd/smoke-test/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,18 @@
+ARG ONOS_BUILD_VERSION=stable
```
Comment (Andrea-Campanella) [inline cmd/smoke-test/Dockerfile]: Just out of curiosity, why are you creating a docker image just for the test ? what does this have that we can't move to the base docker image ?

File: cmd/smoke-test/smoketest_test.go [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+
+	setTZRequest := &gpb.SetRequest{}
+	if err := proto.UnmarshalText(protoPath, setTZRequest); err != nil {
+		return err
+	}
+
+	_, err := c.(*gclient.Client).Set(ctx, setTZRequest)
+	return err
+}
+
```
Comment (Andrea-Campanella) [inline cmd/smoke-test/smoketest_test.go]: I know this is WIP but it's a great start. I'm ok with adding subscribe ONCE, STREAM and POLL in a subsequent patch, or here if you prefer.

File: cmd/smoke-test/smoketest_test.go [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+		Addrs:       q.Addrs,
+		Target:      q.Target,
+		Timeout:     q.Timeout,
+		Credentials: q.Credentials,
+		TLS:         q.TLS,
+	})
+	assert.NilError(t, err)
+	assert.Assert(t, c != nil, "Fetching client returned nil")
+
+	// First lookup should return an error - the path has not been given a value yet
```
Comment (Andrea-Campanella) [inline cmd/smoke-test/smoketest_test.go]: my suggestion to make this modular is to have these as functions down below in the class and store them in an array in this "main" method. Iterating over the array and calling the function woudl execute the tests. anybody who wants to add his own could just writ a function and add it to that array.

**Comment by ray-milkey** [comment]
Subsumed into the integration test framework changes

**Pull request closed without merging**